### ISSUE TITLE
Hogwarts Legacy - Game stuttering

### ISSUE THREAD

**Issue by MorrowxD** [HansKristian-Work/vkd3d-proton#1423]
Game is stuttering while moving to difrent locations, even if I comeback to location where i allready been the game is stutter still. Even on the lowest setings with FSR 2 set on performance game i sttuter, also FPS cap for 60 dosn't change anything.  I tried proton expiremental and GE 7-49, with mesa 22 and 23. Bewllow is my stem spec, is anyone also occured this problem? Is it game related or linux stack related? 

Computer Information:
    Manufacturer:  Gigabyte Technology Co., Ltd.
    Model:  B550 AORUS ELITE AX V2
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  AuthenticAMD
    CPU Brand:  AMD Ryzen 7 5800X 8-Core Processor             

Operating System Version:
    Fedora Linux 37 (Workstation Edition) (64 bit)
    Kernel Name:  Linux
    Kernel Version:  6.1.10-200.fc37.x86_64
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12201007
    X Window Manager:  GNOME Shell
    Steam Runtime Version:  steam-runtime_0.20221019.0

Video Card:
    Driver:  AMD AMD Radeon RX 6800 XT (navi21, LLVM 15.0.7, DRM 3.49, 6.1.10-200.fc37.x86_64)
    Driver Version:  4.6 (Compatibility Profile) Mesa 23.1.0-devel
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 164 Hz
    VendorID:  0x1002
    DeviceID:  0x73bf
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 23.62" x 13.39" (27.13" diag)
                                            60.0cm x 34.0cm (68.9cm diag)
    Primary VRAM: 16384 MB

Sound card:
    Audio device: ATI R6xx HDMI

Memory:
    RAM:  31996 MB

**Comment by Blisto91** [comment]
Is it still the same with the new game update?

**Comment by K0bin** [comment]
This is a problem with the game. It does that on Windows too.

**Comment by MorrowxD** [comment]
> Is it still the same with the new game update?

Yes game is still stuttering even with update.

Is there is any debug tool for vkd3d to listen api calls? Like strace for syscals? Maybe this will help to diagnose the issue. 
Any idea how to debug this? And provide valuable information for this issue?
Even if this is a game issue I'm still want to learn to debug stuff's like that. 




> This is a problem with the game. It does that on Windows too.

Ok i'll install windows on second disk and check it.

**Comment by DistantThunder** [comment]
I'd chance the stuttering in the game usually happens when roaming around close areas the player may visit soon-ish, that the game tries to preload the occluded environment based off proximity and the way the engine does that is heavy on the current scene being drawn.

With the update, I'm speculating again, but I have the impression they made the loading less aggressive, more "just-in-time" leading to some reports of people after the update saying some areas have blank spots for a few seconds upon entering them.

**Comment by MorrowxD** [comment]
I installed Windows and noticed drops in FPS in the same situations, as well as minimal stutters, but the frame time is a lot better, so the stuttering is less noticeable. Therefore, I think there may be something to optimize on Linux.

**Comment by nr1971** [comment]
I've been really annoyed by the stutters in this game and last night I was playing around with different settings and found that using VKD3D_CONFIG=no_upload_hvv works wonders to reduce stutters on my system. (5600x and rx6600xt) 

 There is a small performance hit, maybe 5,10 fps at most but I'm getting 70-100 fps anyway with mostly high settings 1440p and fsr 2 quality so it doesn't really matter. I have resizable BAR and above 4g decoding on in my bios btw. I also have some engine.ini tweaks added that help but the command above makes a huge difference.

**Comment by marco-kretz** [comment]
> I've been really annoyed by the stutters in this game and last night I was playing around with different settings and found that using VKD3D_CONFIG=no_upload_hvv works wonders to reduce stutters on my system. (5600x and rx6600xt)

This helped A LOT to reduce micro-stutters while moving the camera! Thanks mate!

**Comment by mbriar** [comment]
Microstutter when moving the camera isn't something I would expect VKD3D_CONFIG=no_upload_hvv to help with, are you sure it's not the broken-by-default power scaling on rdna2 that maybe no_upload_hvv helps to work around? Try this workaround on it's own, maybe: https://gitlab.freedesktop.org/drm/amd/-/issues/1500#note_825883

**Comment by Blisto91** [comment]
Also. New big update to test hehe https://hogwartslegacy.bugs.wbgames.com/patch-notes

**Comment by marco-kretz** [comment]
> Microstutter when moving the camera isn't something I would expect VKD3D_CONFIG=no_upload_hvv to help with, are you sure it's not the broken-by-default power scaling on rdna2 that maybe no_upload_hvv helps to work around? Try this workaround on it's own, maybe: https://gitlab.freedesktop.org/drm/amd/-/issues/1500#note_825883

Maybe it was just placebo. The workaround def helped as I'm currently using my old 5700xt and waiting for my 6900xt getting repaired...

New patch sound promising as well!

**Comment by nr1971** [comment]
> Microstutter when moving the camera isn't something I would expect VKD3D_CONFIG=no_upload_hvv to help with, are you sure it's not the broken-by-default power scaling on rdna2 that maybe no_upload_hvv helps to work around? Try this workaround on it's own, maybe: https://gitlab.freedesktop.org/drm/amd/-/issues/1500#note_825883

No, I'm not sure if it's the power scaling. Is corectrl doing what those commands do? I have a gaming profile setup for high performance with it. Also, I would  think that stutters would be present in all my other games and there is not.

**Comment by Blisto91** [comment]
Are you still having issues after latest game update?

**Comment by Blisto91** [comment]
@MorrowxD Friendly ping

**Comment by renanwp2** [comment]
> > Microstutter when moving the camera isn't something I would expect VKD3D_CONFIG=no_upload_hvv to help with, are you sure it's not the broken-by-default power scaling on rdna2 that maybe no_upload_hvv helps to work around? Try this workaround on it's own, maybe: https://gitlab.freedesktop.org/drm/amd/-/issues/1500#note_825883
> 
> Maybe it was just placebo. The workaround def helped as I'm currently using my old 5700xt and waiting for my 6900xt getting repaired...
> 
> New patch sound promising as well!

No, it is not placebo. My frame-time was crazy when I was playing the game and, after using this parameter, everything ended. Just some occasional stuttering now, and not while changing the view.
